### ISSUE TITLE
Training the Volumetric model

### ISSUE THREAD

**Issue by samvaf** [karfly/learnable-triangulation-pytorch#21]
Hello,

I have pre-processed the Human3.6M dataset according to your given [manual](https://github.com/karfly/learnable-triangulation-pytorch/blob/master/mvn/datasets/human36m_preprocessing/README.md).
However, when I try to train the volumetric model, I face this problem:

`Exception has occurred: AssertionError
  File "/media/saman/storage_device/ubuntu/learnable-triangulation/mvn/datasets/human36m.py", line 111, in __init__
    assert len(self.keypoints_3d_pred) == len(self)
  File "/media/saman/storage_device/ubuntu/learnable-triangulation/train.py", line 65, in setup_human36m_dataloaders
    crop=config.dataset.train.crop if hasattr(config.dataset.train, "crop") else True,
  File "/media/saman/storage_device/ubuntu/learnable-triangulation/train.py", line 115, in setup_dataloaders
    train_dataloader, val_dataloader, train_sampler = setup_human36m_dataloaders(config, is_train, distributed_train)
  File "/media/saman/storage_device/ubuntu/learnable-triangulation/train.py", line 446, in main
    train_dataloader, val_dataloader, train_sampler = setup_dataloaders(config, distributed_train=is_distributed)
  File "/media/saman/storage_device/ubuntu/learnable-triangulation/train.py", line 485, in <module>
    main(args)`

It seems thet `len(self)=389938` while `len(self.keypoints_3d_pred)=159181`

Thank you for helping me!

**Comment by karfly** [comment]
Hello, @samvaf.
I think this problem appears because of the different number of extracted frames in your and my data. Unfortunately, I can't debug your problem right now, because I'm on the business trip (it'll take ~3 weeks).

I propose 2 possible solutions for now:
1. Just use ground-truth pelvises instead of predicted ones. Change field `use_gt_pelvis: false` to `use_gt_pelvis: true` in the train config. But note, such training mode overestimates the resulting accuracy, because pelvis is always perfectly defined.

2. Precalculate pelvises using pretrained Algebraic model. To do it 1) just run:

```bash
python3 train.py \
  --eval --eval_dataset train \
  --config eval/human36m_alg.yaml \
  --logdir ./logs
```
Then 2) find `results.pkl` file in `./logs` dir and edit field in the `experiments/human36m/train/human36m_vol_softmax.yaml` config:
```yaml
train:
    pred_results_path: {PATH_TO_YOUR_RESULTS}
```
Repeat steps 1) and 2) for `val` dataset.

**Comment by samvaf** [comment]
Hello @karfly,

Thank you for your reply.

If the problem is the number of frames, I am going to double-check the procedure which I have done to pre-process the Human3.6M.

**Comment by dulibubai** [comment]
Hello! @samvaf ,
How did you download the Human3.6M dataset ? I have log in the human net with my and my mentor's emails,but it needs manual verification for a long time,.Until I don't have Human3.6M ,It's very significant for my study research.  
Thanks very much.

**Comment by samvaf** [comment]
Hello @dulibubai,

In fact, I had requested to have access to the Human3.6M a couple of months ago. Yes, you're right, for me it took around one month to have the account confirmed! Just be sure that you have requested with your academic email, not your personal email!

**Comment by dulibubai** [comment]
@samvaf,Thanks a lot!

**Comment by myshli** [comment]
@samvaf Hey! Any chance you can provide us a small subset of Human3.6M? We need just one "experiment" to understand what kind of data the model expects as an input. We are trying to make the model run on our footage, but it`s really hard to do without an example of data it was trained on. And we`ve been waiting for approval by Human3.6M admins for almost 3 weeks already and there are no signs of activity on their end :(

**Comment by dulibubai** [comment]
@myshli, Hi!,you can take the source code about processing Human3.6M dataset to judge the form of Human3.6M. I dont't have the dataset ,either. I am trying do so! It's not hard.
